src/range.rs

```diff
@@ -202,23 +202,40 @@ impl<V: Ord> Range<V> {
     /// Returns true if the this Range contains the specified value.
     pub fn contains(&self, v: &V) -> bool {
         for segment in self.segments.iter() {
-            if match segment {
-                (Unbounded, Unbounded) => true,
-                (Unbounded, Included(end)) => v <= end,
-                (Unbounded, Excluded(end)) => v < end,
-                (Included(start), Unbounded) => v >= start,
-                (Included(start), Included(end)) => v >= start && v <= end,
-                (Included(start), Excluded(end)) => v >= start && v < end,
-                (Excluded(start), Unbounded) => v > start,
-                (Excluded(start), Included(end)) => v > start && v <= end,
-                (Excluded(start), Excluded(end)) => v > start && v < end,
-            } {
+            if !within_lower_bound(segment, v) {
```

Comment:
Ah ok, I I'm getting it now. The naming sounds a bit weird? "within bounds" makes sense, but "within lower bound" feels odd. I'd say something like "above lower bound", "below upper bound". Or if we want to avoid the negation, use "below" in both cases, like this:
```rust
if below_lower_bound( ... ) {
  return false;
else if below_upper_bound( ... ) {
  return true;
}
```

PS, there is a typo in `within_uppern_bound` with an "n" at the end of "upper".

Reply:
Or alternatively, a method on the bounded range maybe:
```rust
if segment.is_above(v) {
  return false;
} else if !segment.is_below(v) {
  return true;
}
```
I'm not a super fan of the not `!` in there though.

Reply:
How about a `within_bounds` that returns a https://doc.rust-lang.org/std/cmp/enum.Ordering.html ?

Reply:
So taking `Ordering` as a slightly richer `Bool`. It's tempting. In that case it's important to have the version argument before the segment argument (`within_bounds(v, segment)`) because ordering can be confusing.